content-src/components/DiscoveryStreamComponents/CardGrid/CardGrid.jsx

```diff
@@ -12,23 +13,40 @@ export class CardGrid extends React.PureComponent {
       );
     }
 
-    let cards = data.recommendations.slice(0, this.props.items).map((rec, index) => (
-      <DSCard
-        key={`dscard-${index}`}
-        pos={rec.pos}
-        campaignId={rec.campaign_id}
-        image_src={rec.image_src}
-        title={rec.title}
-        excerpt={rec.excerpt}
-        url={rec.url}
-        id={rec.id}
-        type={this.props.type}
-        context={rec.context}
-        dispatch={this.props.dispatch}
-        source={rec.domain}
-        pocket_id={rec.pocket_id}
-        bookmarkGuid={rec.bookmarkGuid} />
-    ));
+    if (!data.recommendations.length) {
```

Comment:
I think rendering the `ds-header` in both the empty and non-empty case is probably not the best, maybe you could do something like this?
```js
render() {
    const {data} = this.props;

    // Handle a render before feed has been fetched by displaying nothing
    if (!data) {
      return null;
    }

    // Handle the case where a user has dismissed all recommendations
    const isEmpty = data.recommendations.length === 0;

    return (<div>
      <div className="ds-header">{this.props.title}</div>
      {isEmpty ?
        <div className="ds-card-grid empty"><DSEmptyState /></div> :
        this.renderCards()
      }
    </div>);
  }
}
```